catchup/service.go

```diff
@@ -492,6 +511,65 @@ func (s *Service) sync() {
 	s.log.Infof("Catchup Service: finished catching up, now at round %v (previously %v). Total time catching up %v.", s.ledger.LastRound(), pr, elapsedTime)
 }
 
+// TODO this doesn't actually use the digest from cert!
+func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.AsyncVoteVerifier) {
```

Comment:
Instead of passing the `AsyncVoteVerifier`, perhaps you can use the `BlockAuthenticator` that you already have?

Reply:
The BlockAuthenticator is using lowPriorityCryptoVerificationPool whereas the passed-in verifier is using a highPriorityCryptoVerificationPool.

I agree that it's a bit clunky, and should be passed only on initialization. Let's do this cleanup in a follow-up change ?

Reply:
Sounds good.